[10.0] migrate delivery_carrier_label postlogistics

yvaucher opened the pull request:
cherry pick of commits from 9.0

Plus

* Simplify signature of soap connector by removing company
* As carrier inherits from product, change postlogistics partner as
  a service product

Specific changes to v 10.0 are in https://github.com/yvaucher/carrier-delivery/commit/abfa3395b1666f41b7673f4f24d8878a6e5f2dd3

yvaucher committed:
Add test mode using integration server of Swiss post + however, it needs https://fedorahosted.org/suds/attachment/ticket/239/suds_recursion.patch on suds library

Conflicts:
	delivery_carrier_label_postlogistics/company.py
	delivery_carrier_label_postlogistics/postlogistics/web_service.py
	delivery_carrier_label_postlogistics/res_config.py
	delivery_carrier_label_postlogistics/res_config_view.xml

yvaucher committed:
Add tracking number format option.

By default let postlogistics generate the tracking number.

The new option is to use picking number or concatenation of picking
number and pack counter.

Conflicts:
	delivery_carrier_label_postlogistics/company.py
	delivery_carrier_label_postlogistics/postlogistics/web_service.py
	delivery_carrier_label_postlogistics/res_config.py
	delivery_carrier_label_postlogistics/res_config_view.xml

yvaucher committed:
Implement delivery instructions ZAW

yvaucher committed:
Improve error handling for 3 default parameters layout, output format and resolution

yvaucher committed:
Refactor generation of tracking number to make it overridable

yvaucher committed:
Don't send unset data for delivery instruction in order to get the error message from webservice

yvaucher committed:
Add a note on patch to apply + a copy of it in the module

yvaucher committed:
Add nocache with suds to force the reload of the definition otherwise it
work once then fail on next request.

yvaucher committed:
Concatenate all pack tracking numbers on picking carrier tracking ref field when using packs

yvaucher committed:
Add option to print ProClima logo

yvaucher committed:
Use configuration_helper to simplify the settings class (#84)

yvaucher committed:
Fix assignation of tracking ref

yvaucher committed:
Fix errors on check of parameters

yvaucher committed:
[MIG] delivery_carrier_label_postlogistics to 10.0

* Simplify signature of soap connector by removing company
* As carrier inherits from product, add a service postlogistics product
* Use an image instead of base64 for postlogistics logo

yvaucher committed:
Fix missing decorator on fields_view_get

yvaucher committed:
Fix tests with change of xmlid

grindtildeath commented on delivery_carrier_label_postlogistics/models/delivery.py:
```diff
@@ -105,29 +105,25 @@ class DeliveryCarrierOption(models.Model):
                     node.set(
                         'domain',
-                        "[('partner_id', '=', %s), "
+                        "[('product_id', '=', %s), "
```
There is no `product_id` on `delivery.carrier.template.option` model, what brings :
```
ValueError: Invalid field u'product_id' in leaf "<osv.ExtendedLeaf: (u'product_id', u'=', 2) on delivery_carrier_template_option (ctx: )>"
```

grindtildeath committed:
Fix domain on delivery.carrier.option tmpl_option_id_domain field. Use delivery_type instead of carrier_type.

grindtildeath committed:
Use partner_id instead of product to create template options

grindtildeath committed:
Define postlogistics_ methods

grindtildeath committed:
Add to contributors

Fix travis

Improvements after review

Define new readme folder for generator

Fix manifest, import, rst, etc

simahawk commented on delivery_carrier_label_postlogistics/__manifest__.py:
```diff
@@ -2,23 +2,26 @@
+ 'depends': ['base_delivery_carrier_label',
+             'configuration_helper',
+             'web_domain_field'],
  'website': 'http://www.camptocamp.com/',
```
ttps://github.com/OCA/delivery-carrier

simahawk commented on delivery_carrier_label_postlogistics/data/partner.xml:
```diff
@@ -0,0 +1,10 @@
+<odoo>
+  <record id="partner_postlogistics" model="res.partner">
```
noupdate?

simahawk commented on delivery_carrier_label_postlogistics/models/company.py:
```diff
@@ -51,11 +53,28 @@ class ResCompany(models.Model):
-    def _get_wsdl_url(self):
+    @api.depends('postlogistics_test_mode')
+    def _compute_wsdl_url(self):
+        path = 'delivery_carrier_label_postlogistics/data/'
```
would be better to use `from odoo.modules.module import get_resource_path`

simahawk commented on delivery_carrier_label_postlogistics/README.rst:
```diff
@@ -23,6 +23,15 @@ As a requirement you need to install `suds-jurko` library. It will fail with the
 latest and outdated version of `suds`.
 https://pypi.python.org/pypi/suds-jurko/0.6
```
can you update whole readme based on new layout? https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

simahawk reviewed (changes requested):
small things

grindtildeath commented:
@simahawk Fixed, and squashed my last commits. I left @yvaucher 's as they are.

simahawk commented:
@grindtildeath once you fix this warning on runbot we should be able to merge /cc @gurneyalex @yvaucher 
```
2018-05-29 11:16:29,382 182 WARNING openerp_test odoo.addons.base.module.module: module delivery_carrier_label_postlogistics: description is empty !
```

grindtildeath committed:
Automatically patch suds library on module loading

This allows to use suds library without having to patch manually for recursion issue.
Moreover it logs any HTTP error and displays it with a meaningful error.

grindtildeath commented:
@simahawk Shouldn't we backport this commit ? https://github.com/OCA/delivery-carrier/pull/155/commits/64e34224d6754e5cb65e3ad8eb73b02526b76545

simahawk commented:
@grindtildeath I'd say "yes". See my comments on it ;)
